back-end/tests/test_predict.py

```diff
@@ -3,77 +3,45 @@
-            )
+            assert rv['error_code'] == -1
+            assert rv['detail'] == 'Invalid image path ' + RServer.getServer().baseDir + '/dataset/train/bird/10000.JPEG'
             # TODO: [test] other splits
 
         def test_predict_success(self, client):
-            response = client.get(
-                "/predict/train?"
-                + PARAM_NAME_IMAGE_PATH
-                + "="
-                + RServer.getServer().baseDir
-                + "/dataset/train/bird/1.JPEG"
-            )
+            response = client.get("/predict/train?" + PARAM_NAME_IMAGE_PATH +
+                                  "=" + RServer.getServer().baseDir + "/dataset/train/bird/1.JPEG")
             assert response.status_code == 200
             rv = response.get_json()
-            assert rv["code"] == 0
-            data = rv["data"]
-            assert data[0] == [
-                "bird",
-                "cat",
-                "crab",
-                "dog",
-                "fish",
-                "frog",
-                "insect",
-                "primate",
-                "turtle",
-            ]
+            assert rv['code'] == 0
+            data = rv['data']
+            assert data[0] == ['bird', 'cat', 'crab', 'dog', 'fish',
+                               'frog', 'insect', 'primate', 'turtle']
             assert len(data[1]) == 9
             assert sum((0 <= x <= 1) for x in data[1]) == 9
             assert data[2] == [
-                RServer.getServer().baseDir
-                + "/visualize_images/"
-                + to_snake_path(RServer.getServer().baseDir)
-                + "_dataset_train_bird_1_JPEG_0.png",
-                RServer.getServer().baseDir
-                + "/visualize_images/"
-                + to_snake_path(RServer.getServer().baseDir)
-                + "_dataset_train_bird_1_JPEG_1.png",
-                RServer.getServer().baseDir
-                + "/visualize_images/"
-                + to_snake_path(RServer.getServer().baseDir)
-                + "_dataset_train_bird_1_JPEG_2.png",
-                RServer.getServer().baseDir
-                + "/visualize_images/"
-                + to_snake_path(RServer.getServer().baseDir)
-                + "_dataset_train_bird_1_JPEG_3.png",
-            ]
+                RServer.getServer().baseDir + '/visualize_images/' +
+                to_snake_path(RServer.getServer().baseDir) + '_dataset_train_bird_1_JPEG_0.png',
+                RServer.getServer().baseDir + '/visualize_images/' +
+                to_snake_path(RServer.getServer().baseDir) + '_dataset_train_bird_1_JPEG_1.png',
+                RServer.getServer().baseDir + '/visualize_images/' +
+                to_snake_path(RServer.getServer().baseDir) + '_dataset_train_bird_1_JPEG_2.png',
+                RServer.getServer().baseDir + '/visualize_images/' +
+                to_snake_path(RServer.getServer().baseDir) + '_dataset_train_bird_1_JPEG_3.png']
```

Comment:
Consider using a list comprehension and string template here? 
Something like `[f"{RServer.getServer().baseDir}/visualize_images/{to_snake_path(RServer.getServer().baseDir)}_ _dataset_train_bird_1_JPEG_{idx}.png" for idx in range(4)]`

Reply:
good idea! Makes the code more readable and concise. Will apply.